test/custom-parser.3.test.js

```diff
@@ -200,57 +183,38 @@ test('catch all content type parser should not interfere with content type parse
     })
   })
 
-  fastify.listen({ port: 0 }, err => {
-    t.assert.ifError(err)
-
-    let pending = 3
+  const fastifyServer = await fastify.listen({ port: 0 })
 
-    function completed () {
-      if (--pending === 0) {
-        done()
-      }
+  const result1 = await fetch(fastifyServer, {
```

Comment:
I think the scope of this test was to run the request concurrently instead of serially as this test is now doing

Could you change it to a `Promise.all`?